From 88c1fc23bb22d0935224f8bb4f67a19de17332a4 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@cloudera.com>
Date: Mon, 9 Jan 2012 14:35:00 -0800
Subject: [PATCH 037/795] MR1: CLOUDERA BUILD. fixing MR1 TestSubmitJob failure.

  Author: Alejandro Abdelnur
  Ref: CDH-4066
(cherry picked from commit 7d9dd4fb5ebd8a0465c0558e3678935a77fc9409)

(cherry picked from commit 1d234d22b13b2b60720cf170d61e9bf83972a1d8)
---
 .../org/apache/hadoop/mapred/TestSubmitJob.java    |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/test/org/apache/hadoop/mapred/TestSubmitJob.java b/src/test/org/apache/hadoop/mapred/TestSubmitJob.java
index a623839..5f38c8d 100644
--- a/src/test/org/apache/hadoop/mapred/TestSubmitJob.java
+++ b/src/test/org/apache/hadoop/mapred/TestSubmitJob.java
@@ -198,7 +198,8 @@ static JobSubmissionProtocol getJobSubmitClient(JobConf conf,
   public void testSecureJobExecution() throws Exception {
     LOG.info("Testing secure job submission/execution");
     MiniMRCluster mr = null;
-    Configuration conf = new Configuration();
+    JobConf conf = new JobConf();
+    conf.set("mapreduce.jobtracker.staging.root.dir", "/tmp/staging");
     final MiniDFSCluster dfs = new MiniDFSCluster(conf, 1, true, null);
     try {
       FileSystem fs =
@@ -215,7 +216,7 @@ public FileSystem run() throws IOException {
               "/tmp/hadoop/mapred/staging"));
       UserGroupInformation MR_UGI = UserGroupInformation.getLoginUser();
       mr = new MiniMRCluster(0, 0, 1, dfs.getFileSystem().getUri().toString(),
-          1, null, null, MR_UGI);
+          1, null, null, MR_UGI, conf);
       JobTracker jt = mr.getJobTrackerRunner().getJobTracker();
       String jobTrackerName = "localhost:" + mr.getJobTrackerPort();
       // cleanup
-- 
1.7.0.4

